src/bgw/job_stat.c

```diff
@@ -660,18 +670,28 @@ ts_bgw_job_stat_mark_end(BgwJob *job, JobResult result)
 								  &res,
 								  ShareRowExclusiveLock))
 		elog(ERROR, "unable to find job statistics for job %d", job->fd.id);
+
+	/* Failures are always logged */
+	if (ts_guc_enable_job_execution_logging || result != JOB_SUCCESS)
+		ts_bgw_job_stat_history_mark_end(job, result, edata);
+
 	pgstat_report_activity(STATE_IDLE, NULL);
 }
 
 void
-ts_bgw_job_stat_mark_crash_reported(int32 bgw_job_id)
+ts_bgw_job_stat_mark_crash_reported(BgwJob *job, JobResult result, Jsonb *edata)
 {
-	if (!bgw_job_stat_scan_job_id(bgw_job_id,
+	if (!bgw_job_stat_scan_job_id(job->fd.id,
 								  bgw_job_stat_tuple_mark_crash_reported,
 								  NULL,
 								  NULL,
 								  RowExclusiveLock))
-		elog(ERROR, "unable to find job statistics for job %d", bgw_job_id);
+		elog(ERROR, "unable to find job statistics for job %d", job->fd.id);
```

Comment:
Should this be `Ensure`? Not really related to this PR, but I think we should generally switch to using either `Ensure` or `ereport` and forbid `elog`, to reduce the cases where we wrongly use `XX000` for user or environment errors.

Reply:
Sure, changed `elog` to `ereport`! Thanks!